vimeo-networking/src/main/java/com/vimeo/networking/model/VimeoAccount.java

```diff
@@ -39,53 +40,65 @@
 public class VimeoAccount implements Serializable {
 
     private static final long serialVersionUID = -8341071767843490585L;
-    //    private static final String TOKEN_TYPE_BEARER = "bearer";
 
+    @Nullable
     @GsonAdapterKey("access_token")
     public String accessToken;
+
+    @Nullable
     @GsonAdapterKey("token_type")
     public String tokenType;
+
+    @Nullable
     @GsonAdapterKey("scope")
     public String scope;
+
+    @Nullable
     @GsonAdapterKey("user")
     public User user;
     private String userJSON;
 
-    public VimeoAccount(){
+    public VimeoAccount() {
         //constructor for stag TypeAdapter generation
     }
 
     public VimeoAccount(@Nullable String accessToken) {
         this.accessToken = accessToken;
     }
 
-    public VimeoAccount(String accessToken, String tokenType, String scope, String userJSON) {
-        if (accessToken == null || accessToken.isEmpty() || tokenType == null ||
-            tokenType.isEmpty() || scope == null || scope.isEmpty()) {
+    public VimeoAccount(@Nullable String accessToken, @NotNull String tokenType, @NotNull String scope,
+                        String userJSON) {
+        if ((accessToken != null && accessToken.isEmpty()) || tokenType == null || tokenType.isEmpty() ||
```

Comment:
It seems odd to have the annotation that the access token is nullable but then throw an exception if it is null. 

Also, since you are doing the null plus isEmpty check so many times, it would be good to break this out into a helper somewhere.